### PR TITLE
implement checkboxes to mark task as completed

### PR DESCRIPTION
**Pull request by mariiaovs**

### CODE REVIEW

File: pages/tasks/[id]/index.js [hunk 1]
```diff
@@ -40,6 +41,7 @@ export default function DetailsPage({
           setShowModal={setShowModal}
           onDelete={onDelete}
           onCancel={onCancel}
+          handleCheckboxChange={handleCheckboxChange}
```
Comment (MerleMak) [inline pages/tasks/[id]/index.js:44]:
```suggestion
          onCheckboxChange={onCheckboxChange}
```

File: pages/tasks/[id]/index.js [hunk 2]
```diff
@@ -21,6 +21,7 @@ export default function DetailsPage({
   setShowModal,
   onDelete,
   onCancel,
+  handleCheckboxChange,
```
Comment (MerleMak) [inline pages/tasks/[id]/index.js:24]:
```suggestion
  onCheckboxChange,
```

File: pages/_app.js [hunk 3]
```diff
@@ -49,6 +58,7 @@ export default function App({ Component, pageProps }) {
         showModal={showModal}
         onDelete={handleDeleteTask}
         onCancel={closeModalWindow}
+        handleCheckboxChange={handleCheckboxChange}
```
Comment (MerleMak) [inline pages/_app.js:61]:
```suggestion
        onCheckboxChange={handleCheckboxChange}
```

File: components/TasksList.js [hunk 4]
```diff
@@ -28,8 +32,12 @@ export default function TasksList({ tasks }) {
         <StyledListItems
           key={task.id}
           $current={handleCurrentTask(task.dueDate)}
+          $isDone={task.isDone}
         >
-          <TaskPreview task={task} />
+          <TaskPreview
+            task={task}
+            handleCheckboxChange={handleCheckboxChange}
```
Comment (MerleMak) [inline components/TasksList.js:39]:
```suggestion
            onCheckboxChange={onCheckboxChange}
```

File: components/TasksList.js [hunk 5]
```diff
@@ -11,13 +11,17 @@ const StyledList = styled.ul`
...
+    $isDone ? "lightgray" : $current ? "#f1afaffc" : "white"};
+  opacity: ${({ $isDone }) => $isDone && "0.5"};
+  color: ${({ $isDone }) => $isDone && "gray"};
   padding: 1rem;
   box-shadow: 5px 5px 15px 5px rgba(112, 107, 91, 0.83);
   margin: 0.5rem;
 `;
 
-export default function TasksList({ tasks }) {
+export default function TasksList({ tasks, handleCheckboxChange }) {
```
Comment (MerleMak) [inline components/TasksList.js:24]:
```suggestion
export default function TasksList({ tasks, onCheckboxChange }) {
```

File: components/TaskPreview.js [hunk 6]
```diff
@@ -10,17 +24,22 @@ const StyledParagraph = styled.p`
...
+export default function TaskPreview({ task, handleCheckboxChange }) {
+  const { title, category, priority, dueDate, id, isDone } = task;
 
   return (
     <StyledSection>
-      <Link href={`tasks/${id}`}>
+      <StyledCheckbox
+        type="checkbox"
+        checked={isDone}
+        onChange={() => handleCheckboxChange(id)}
```
Comment (MerleMak) [inline components/TaskPreview.js:35]:
```suggestion
        onChange={() => onCheckboxChange(id)}
```

File: components/TaskPreview.js [hunk 7]
```diff
@@ -10,17 +24,22 @@ const StyledParagraph = styled.p`
   text-align: center;
 `;
 
-export default function TaskPreview({ task }) {
-  const { title, category, priority, dueDate, id } = task;
+export default function TaskPreview({ task, handleCheckboxChange }) {
```
Comment (MerleMak) [inline components/TaskPreview.js:27]:
```suggestion
export default function TaskPreview({ task, onCheckboxChange }) {
```

File: components/TaskDetails.js [hunk 8]
```diff
@@ -63,14 +86,27 @@ export default function TaskDetails({
...
+        <h2>{"🔥".repeat(priority)}</h2>
+        <p>Due Date:</p>
+        <h3>{dueDate}</h3>
+        <label htmlFor="checkbox">
+          Done:
+          <StyledCheckbox
+            type="checkbox"
+            id="checkbox"
+            checked={isDone}
+            onChange={() => handleCheckboxChange(id)}
```
Comment (MerleMak) [inline components/TaskDetails.js:106]:
```suggestion
            onChange={() => onCheckboxChange(id)}
```

File: components/TaskDetails.js [hunk 9]
```diff
@@ -63,14 +86,27 @@ export default function TaskDetails({
...
-    </StyledSection>
+      <StyledSection $isDone={isDone}>
+        <StyledTrash onClick={() => setShowModal(true)} />
+
+        <p> What is to do?</p>
+        <h2>{title}</h2>
+        <p>Category: </p>
+        <h2>{category}</h2>
+        <p>Priority: </p>
+        <h2>{"🔥".repeat(priority)}</h2>
```
Comment (MerleMak) [inline components/TaskDetails.js:97]: cool!!

File: components/TaskDetails.js [hunk 10]
```diff
@@ -38,23 +51,33 @@ const StyledPragraph = styled.p`
...
+  }
+`;
+
 export default function TaskDetails({
   task,
   showModal,
   setShowModal,
   onDelete,
   onCancel,
+  handleCheckboxChange,
```
Comment (MerleMak) [inline components/TaskDetails.js:70]:
```suggestion
  onCheckboxChange,
```

**Review by MerleMak — COMMENTED** [review]
super nice feature!

I left you some comments only concerning a naming convention :)
usually functions are called with the prefix "on" and passed as props with this naming up until the place in the app where the functions in defined. only there (in your case its in the _app.js) the function in named with the prefix "handle" :) 

Otherwise you did a great job, keep it up!!

**Review by SwethaJanardhana — APPROVED** [review]
LGTM

**Review by SwethaJanardhana — APPROVED** [review]
LGTM

**Pull request merged**